### PR TITLE
Re-write mau-qa_userspace_nfs(v4) test cases

### PR DESCRIPTION
**Pull request by rfan1**
https://progress.opensuse.org/issues/99636
https://progress.opensuse.org/issues/124541

- Verification run: [VR](http://openqa.suse.de/tests/overview?build=nfs_qa_run_renew&distri=sle)

### CODE REVIEW

**Comment by dzedro** [comment]
Looks to do more than just sync, as you already looked into `/usr/share/qa/qa_test_nfs/nfs_read_write_fn.sh` it should be done IMO in https://github.com/SUSE/qa-testsuites/tree/master/tests/qa_test_nfs :smile:

**Comment by rfan1** [comment]
> Looks to do more than just sync, as you already looked into `/usr/share/qa/qa_test_nfs/nfs_read_write_fn.sh` it should be done IMO in https://github.com/SUSE/qa-testsuites/tree/master/tests/qa_test_nfs smile

As we discussed, Let me re-write this test code and don't run it with qa_run.pm.

I will extracted all test scripts from the package 'qa_test_nfs' and add the workaround to the script as well

**Comment by rfan1** [comment]
> > Looks to do more than just sync, as you already looked into `/usr/share/qa/qa_test_nfs/nfs_read_write_fn.sh` it should be done IMO in https://github.com/SUSE/qa-testsuites/tree/master/tests/qa_test_nfs smile
> 
> I failed to open the link above, is there any permission control there?

Now I use a simple method to re-use the shell scripts directly

File: tests/network/nfs.pm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use utils;
+use version_utils 'is_sle';
+
+sub nfs_test {
+    my ($nfs_ver) = @_;
+    $nfs_ver //= 'nfs';
+    my @test_cases
+      = qw(nfs_start_stop nfs_mount_umount nfs_read nfs_write nfs_dontwrite nfs_usermapping_rootsquash nfs_usermapping_norootsquash nfs_usermapping_allsquash);
+    foreach my $test_case (@test_cases) {
+        record_info("$nfs_ver: $test_case Starts");
```
Comment (dzedro) [inline tests/network/nfs.pm:35]: I would remove the `Starts`, it makes the text just longer and harder to read
Generally the record info is at the beginning of the "operation".

**Review by dzedro — APPROVED** [review]

File: tests/network/nfs.pm (hunk 1 above)
Comment (rfan1) [reply tests/network/nfs.pm:35]: Make sense, will do

**Pull request merged**